From 3fe00ce1f18035f77abb8a0a1e62200be5735a2a Mon Sep 17 00:00:00 2001
From: Aaron Kling <webgeek1234@gmail.com>
Date: Thu, 14 Dec 2017 15:14:08 -0600
Subject: [PATCH 03/22] Include build manifest on target

Private repos can be excluded by setting the env var
MANIFEST_EXCLUDES | delimited. proprietary is always
excluded.

Change-Id: I9a085a7102d9905853636e0ad60ac6f80a930ae0
---
 build/tasks/build_manifest.mk | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)
 create mode 100644 build/tasks/build_manifest.mk

diff --git a/build/tasks/build_manifest.mk b/build/tasks/build_manifest.mk
new file mode 100644
index 00000000..ee89f8c8
--- /dev/null
+++ b/build/tasks/build_manifest.mk
@@ -0,0 +1,29 @@
+# Copyright (C) 2017 The LineageOS Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+BUILD_MANIFEST := $(TARGET_OUT_ETC)/build-manifest.xml
+
+ifdef MANIFEST_EXCLUDES
+MANIFEST_EXCLUDES := |$(MANIFEST_EXCLUDES)
+endif
+
+# Regenerate the manifest on every full build
+.PHONY: FORCE
+FORCE:
+
+$(BUILD_MANIFEST): FORCE
+	mkdir -p $(dir $@)
+	repo manifest -o - -r | grep -Ev "proprietary_$(MANIFEST_EXCLUDES)" > $@
+
+$(FULL_SYSTEMIMAGE_DEPS): $(BUILD_MANIFEST)
-- 
2.17.1

